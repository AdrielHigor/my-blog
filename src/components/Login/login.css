form#Login {
    width: 100%;
    max-width: 450px;
    margin: 30px auto 0;
    padding: 30px;
    background: #ffffff;
    border: 1px solid #dddddd;
    display: flex;
    flex-direction: column;
}

form#Login input{
    margin-bottom: 15px;
    height: 40px;
    border-radius: 4px;
    border: 1px solid #ddd;
    padding: 0 20px;
    font-size: 14px;
}

form#Login button{
    padding:10px 20px;
    border-radius: 4px;
    background: #25282a;
    color: #ffffff;
    font-size: 18px;
    font-weight: bold;
    cursor: pointer;
}

form#Login a, p{
    text-align: center;
    text-decoration: none;
    font-size: 16px;
    padding-top: 20px;
}

form#Login a{
    font-weight: bold;
}