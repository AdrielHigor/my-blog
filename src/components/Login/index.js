import "./login.css";
import React, { useEffect } from "react";
import { Link, withRouter } from "react-router-dom";
import firebase from "../../firebase";

export default withRouter(function Login(props) {
    const [email, setEmail] = React.useState();
    const [password, setPassword] = React.useState();

    const login = React.useCallback(
        async e => {
            e.preventDefault();

            try {
                await firebase.login(email, password).catch(error => {
                    alert(error.code);
                    return null;
                });
                props.history.replace("/dashboard");
            } catch (error) {
                alert(error.message);
            }
        },
        [email, password]
    );

    useEffect(() => {
        if (firebase.getCurrent()) {
            props.history.replace('/dashboard')
        }
    })

    return (
        <div>
            <form onSubmit={login} id="Login">
                <label>Email: </label>
                <input
                    type="email"
                    autoFocus
                    value={email}
                    onChange={e => setEmail(e.target.value)}
                    placeholder="email@email.com"
                />
                <label>Senha: </label>
                <input
                    type="password"
                    value={password}
                    onChange={e => setPassword(e.target.value)}
                    placeholder="*******"
                />
                <button type="submit">Entrar</button>
                <p>
                    Ainda nÃ£o tem conta? <Link to="/register"> Cadastre-se </Link>
                </p>
            </form>
        </div>
    );
});
