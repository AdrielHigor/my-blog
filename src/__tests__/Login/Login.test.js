import React from "react";
import Login from "../../components/Login";
import { shallow } from "enzyme";
import "../../setupTests";

describe("Login Test Suite", () => {
    test("Login rendering correctly", () => {
        const component = shallow(<Login.WrappedComponent />)

        expect(component).toMatchSnapshot();
    })

    test("Login submit form correctly", () => {
        let component
        jest
            .spyOn(React, 'useState')
            .mockImplementationOnce(() => ["testemail@teste.com", jest.fn()])
            .mockImplementationOnce(() => ["passw0rd", jest.fn()])

        const mockLogin = jest
            .spyOn(React, 'useCallback')
            .mockImplementation(() => { })

        component = shallow(<Login.WrappedComponent />);

        component.find("button").simulate("click");

        expect(mockLogin).toHaveBeenCalled();
    })
})