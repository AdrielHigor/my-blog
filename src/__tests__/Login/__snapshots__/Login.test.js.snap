// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Test Suite Login rendering correctly 1`] = `ShallowWrapper {}`;
